#!/bin/sh -e
## calculate-average.dpatch by Dean Gaudet <dean@arctic.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes --calculate-average
## DP: Backported from CVS
## DP: http://savannah.nongnu.org/cgi-bin/viewcvs/rdiff-backup/rdiff-backup/rdiff_backup/Main.py.diff?r1=1.74&r2=1.75&diff_format=u

if [ $# -lt 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
       -patch) patch $patch_opts -p0 < $0;;
       -unpatch) patch $patch_opts -p0 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

@DPATCH@
--- rdiff_backup/Main.py	2004/05/15 23:31:24	1.74
+++ rdiff_backup/Main.py	2004/05/16 20:50:33	1.75
@@ -178,7 +178,8 @@
 		if l == 2: pass # Will determine restore or backup later
 		else:
 			commandline_error("Switches missing or wrong number of arguments")
-	elif action == 'test-server': pass # test-server takes any number of args
+	elif action == 'test-server' or action == 'calculate-average':
+		pass # these two take any number of args
 	elif l > 2 or action not in arg_action_dict[l]:
 		commandline_error("Wrong number of arguments given.")
